### DISCUSSION TITLE
composer command

### DISCUSSION THREAD

**Discussion by viniciusbig** [shinsenter/php#14] [General]
I'm using your CakePHP4 image and its awesome!

But with the default image, the composer command will not work

```
docker exec -it <container_id> composer <arguments>
```

Thats because `WORKDIR` is setted to `/` instead of $WEBHOME

There is no composer.json in the root folder.

I've fixed this by creating my own image based on yours but I think this should be the default for all PHP images.
The workdir should probably always point to WEBHOME variable

What do you think?

**Comment by shinsenter** [comment]
@viniciusbig 

Thank you for the suggestion.

In my Dockerfile you can see I have commented out the `WORKDIR` lines.
https://github.com/shinsenter/php/blob/002001c1ef98532cb761eca994ca133ab1727ef7/src/webapps/cakephp4/Dockerfile#L24

I also have a small script to change the correct working directory when user starts `bash` inside the container.
https://github.com/shinsenter/php/blob/002001c1ef98532cb761eca994ca133ab1727ef7/src/base-ubuntu/root/root/.bashrc#L71-L74

Since the default path to the `$WEBHOME` can be changed later through the environment variable, I think it's best for users to set their own WORKDIR in their Dockerfile or docker-compose.yml later.

Regarding the composer issue you mentioned, I will find out later.
It would be great to get more detailed solution from you to be flexible for all the different `$WEBHOME` settings

Best regards.

**Reply by viniciusbig** [reply]
On my setup here, I've created my own Dockerfile adding to install o few other things I miss in your base image.

I ended up adding this line there
```
WORKDIR $WEBHOME
```

So, even if I change `$WEBHOME` to something else, the `WORKDIR` would be correct.

About the script you mention, I didn't notice before.
But is seems its not running because composer command is not working (using the default image)

For example: 
```
docker exec -it <container_id> composer install
```
Will complain that there is no `composer.json` available. (because its being running on root, not WEBHOME)

**Reply by shinsenter** [reply]
@viniciusbig 
I am quite busy with my KPI so I cannot recheck the issue with composer. I will check this issue as soon as I have free time.

**Comment by shinsenter** [comment]
@viniciusbig 
The default working directory is set to /var/www/html in the latest docker images, so I think your problem will be solved.
